### ISSUE TITLE
[Feature Reqeust] Mirror all submodules recursively

### ISSUE THREAD

**Issue by duanshiqiang** [orrc/git-webhook-proxy#7]
Currently this program doesn't mirror any submodule which sometimes is the largest compared with the main repo.

Supporting mirror all submodules would be nice.

Some reference: https://support.cloudbees.com/hc/en-us/articles/115001728812-Using-a-Git-reference-repository

**Comment by orrc** [comment]
Yeah, I remember thinking about adding this.  But, the last time I checked (which was probably 2015), the Git plugin for Jenkins did not use the reference repo for submodules.  So while I agree mirroring submodules would be a useful addition to this software, I'm not sure Jenkins would actually use it.

Also, I'm not really maintaining this software any more, so it's not very likely that I will implement this…
